subworkflows/local/clean_assembly.nf

```diff
@@ -19,17 +22,34 @@ workflow CLEAN_ASSEMBLY {
 
     ch_versions = ch_versions.mix(SEQKIT_SEQ.out.versions)
 
-    // TODO: handle multiple reference genomes, which our current system does
-    BLAST_BLASTN(
+    BLAST_BLASTN_HUMAN_PHIX(
         SEQKIT_SEQ.out.fastx,
-        reference_genome
+        humanPhiX_reference_genome
     )
 
-    ch_versions = ch_versions.mix(BLAST_BLASTN.out.versions.first())
+    ch_versions = ch_versions.mix(BLAST_BLASTN_HUMAN_PHIX.out.versions.first())
+
+    matched_contigs = Channel.empty()
+    if ( host_reference_genome != null) {
+        BLAST_BLASTN_HOST(
+            SEQKIT_SEQ.out.fastx,
+            host_reference_genome
+        )
+
+        ch_versions = ch_versions.mix(BLAST_BLASTN_HOST.out.versions.first())
+
+        TXT_COMBINER(
```

Comment:
No need to another process, you can use https://www.nextflow.io/docs/latest/operator.html#collectfile

```
matched_contigs = channel.of( BLAST_BLASTN_HUMAN_PHIX.out.txt, BLAST_BLASTN_HOST.out.txt ).collectFile(storeDir: "{params.outdir}/decontamination/contaminated_contigs.txt", newLine: true)
```